src/backend/src/api/endpoints/deployment.rs

```diff
@@ -89,16 +102,152 @@ impl DeploymentsEndpoints {
 
     pub async fn create_deployment(
         &mut self,
-        calling_principal: &Principal,
+        calling_principal: Principal,
         sdl: String,
     ) -> Result<DeploymentId, ApiError> {
         self.access_control_service
-            .assert_principal_not_anonymous(calling_principal)?;
+            .assert_principal_not_anonymous(&calling_principal)?;
 
-        let user_id = UserId::new(*calling_principal);
+        let parsed_sdl = SdlV3::try_from_str(&sdl)
+            .map_err(|e| ApiError::invalid_argument(&format!("Invalid SDL: {}", e)))?;
+
+        let user_id = UserId::new(calling_principal);
 
-        let deployment = Deployment::new(sdl, user_id);
+        let deployment_id = self
+            .deployments_service
+            .init_deployment(user_id, sdl)
+            .await?;
 
-        self.deployments_service.create_deployment(deployment).await
+        ic_cdk_timers::set_timer(Duration::from_secs(0), move || {
+            ic_cdk::spawn(async move {
+                if let Err(e) =
+                    handle_deployment(calling_principal, parsed_sdl, deployment_id).await
+                {
+                    print(&format!("Error handling deployment: {:?}", e));
+                }
+            });
+        });
+
+        Ok(deployment_id)
     }
 }
+
+async fn handle_deployment(
+    calling_principal: Principal,
+    parsed_sdl: SdlV3,
+    deployment_id: DeploymentId,
+) -> Result<(), ApiError> {
+    let mut deployment_service = DeploymentsService::default();
+    let dseq = handle_create_deployment(calling_principal, parsed_sdl, deployment_id)
+        .await
+        .map_err(|e| {
+            deployment_service
+                .set_failed_deployment(deployment_id)
+                .expect("Failed to set deployment to failed");
+            e
+        })?;
+
```

Comment:
here we have to save the `dseq` in the deployment, so that it can be used later (e.g. in the close deployment)

Reply:
it's saved in the history in DeploymentUpdate::DeploymentCreated(tx_hash, dseq). Where do you want to save it?

Reply:
It would be better to save it in the deployment itself, otherwise you have to go through the history just to find it

Reply:
was thinking to do a getter method for that no? otherwise we can also just save it there

Reply:
maybe having a `akash_data` field that contains data related to akash, like the dseq